### ISSUE TITLE
Investigate if Razor can use full type names in VS' LSP completion descriptions and hovers

### ISSUE THREAD

**Issue by NTaylorMullen** [dotnet/aspnetcore#20827]
In VSCode tooltips for completion items and hovers are of a limited size so to work around this our language server went through the trouble of minimizing typenames for TagHelpers. Aka for a property `Microsoft.AspNetCore.Razor.TagHelpers.EnvironmentTagHelper.Name` => `EnvironmentTagHelper.Name`.

If VS can handle longer tooltips (it should be able to) we should have a language server platform specific setting that we define to make this experience slightly different in the two IDEs.

/cc @ryanbrandenburg

**Comment by ryanbrandenburg** [comment]
Just so I'm clear, you're saying we should add a custom capability for "HandlesLongNames" to `CompletionCapabilities` and `HoverCapabilities`?

**Comment by NTaylorMullen** [comment]
I'd go a little simpler. I'd add a command line arg to the language server to indicate the platform (we'll need this anyways for telemetry) and based on the value update our own option type (not yet created) `RazorLanguageServerOptions` that had something like `GenerateMinimalDescriptions: true` for VSCode.

**Comment by ryanbrandenburg** [comment]
![ExtraLongHover](https://user-images.githubusercontent.com/14987221/79807901-148a7880-8321-11ea-87c3-8aff27051554.png)
 
So I started this out by making the change unconditionally just to see what it looks like in VS, and it looks like above. Personally I find that kind of hard to parse, and I think the reason is the lack of colorization to break up the block. Should we delay this work until we can enable colorization in hovers/completions? Or am I in the minority having trouble with that? Do we have other reasoning about why we might want to do this now anyway?

**Comment by NTaylorMullen** [comment]
@ryanbrandenburg can you post both versions simple and non-simple.

@danroth27 could you make a call on whether we should delay this work?

**Comment by danroth27** [comment]
I find that visually really hard to parse as well. Is there an equivalent C# experience that we can compare this to?

**Comment by ryanbrandenburg** [comment]
A more direct comparison would be against the existing (non-LSP) Razor hover functionality:
![image](https://user-images.githubusercontent.com/14987221/80008338-77dced80-847c-11ea-8393-69b0b44e2197.png)
Also of note here is that they've broken it up by having a new-line between the two types. That certainly breaks them up visually, but I feel like it kind of looses some of the semantic meaning we intend for it to convey.

Kind of an aside, But while looking at the non-LSP behavior in here I noticed this:
![image](https://user-images.githubusercontent.com/14987221/80009042-621bf800-847d-11ea-9ef2-ffd9075b3040.png)
which seems to indicate that either VS (or at least these sub-sections) has a general concept of "wrapping" to reduce window length, which is a bit of a dangerous area (as any automatic fiddling of human-targeted data is).

**Comment by NTaylorMullen** [comment]
@ryanbrandenburg you feel good about tabling this theN?

**Comment by ryanbrandenburg** [comment]
Yeah, I'm going to vote that we mark this "blocked" on getting support for markdown/colorization in hovers and completions.

**Comment by NTaylorMullen** [comment]
This was done indirectly.
